From ba4e68f1adfde1993ff5486cb05a0b7706ecf0d3 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@wisc.edu>
Date: Wed, 30 Sep 2026 11:35:30 -0600
Subject: [PATCH] Fix C++17 compatibility of Config/std.h with libc++

std::mem_fun was removed in C++17. libstdc++ still provides it, but
libc++ (used on macOS) does not, so including ThePEG/Config/std.h fails
to compile.

Only bring std::mem_fun into the ThePEG namespace where it exists.
---
 Config/std.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Config/std.h b/Config/std.h
index 761745d..0e6c709 100644
--- a/Config/std.h
+++ b/Config/std.h
@@ -98,7 +98,10 @@ using std::swap;
 using std::min;
 using std::max;
 using std::mem_fn;
+// std::mem_fun was removed in C++17 and libc++ enforces this
+#if __cplusplus < 201703L || !defined(_LIBCPP_VERSION)
 using std::mem_fun;
+#endif
 using std::sqrt;
 //using std::pow;
 using std::abs;
-- 
2.56.0

